### ISSUE TITLE
LZ4 segfault while salvaging

### ISSUE THREAD

**Issue by agorrod** [wiredtiger/wiredtiger#1800]
test/format triggered a segfault in the LZ4 library. Stack trace:

```
==5692==ERROR: AddressSanitizer: SEGV on unknown address 0x7f8c539c5a60 (pc 0x7f8c579fdb73 sp 0x7ffe6f610320 bp 0x7f8c539cd4da T0)
    #0 0x7f8c579fdb72 (/lib64/liblz4.so.1+0x4b72)
    #1 0x7f8c53dfca45 in wt_lz4_decompress ext/compressors/lz4/lz4_compress.c:146
    #2 0x95acb1 in __wt_bt_read src/btree/bt_io.c:71
    #3 0x9a4740 in __slvg_read src/btree/bt_slvg.c:396
    #4 0x9a10ed in __wt_bt_salvage src/btree/bt_slvg.c:201
    #5 0x8320cc in __wt_salvage src/session/session_salvage.c:43
    #6 0x7ef75b in __wt_schema_worker src/schema/schema_worker.c:60
    #7 0x7f1a7e in __wt_schema_worker src/schema/schema_worker.c:106
    #8 0x80e591 in __session_salvage src/session/session_api.c:551
    #9 0x4cc62e in salvage test/format/salvage.c:47
    #10 0x4cc1b4 in wts_salvage test/format/salvage.c:177
    #11 0x4cf111 in main test/format/t.c:234
    #12 0x37cda21d64 in __libc_start_main (/lib64/libc.so.6+0x37cda21d64)
    #13 0x4a418c in _start (/home/jenkins/jenkins/workspace/wiredtiger-test-format-stress-sanitizer/build_posix/test/format/t+0x4a418c)
```

Configuration:

```
############################################
#  RUN PARAMETERS
############################################
abort=0
auto_throttle=1
firstfit=0
bitcnt=8
bloom=1
bloom_bit_count=18
bloom_hash_count=22
bloom_oldest=1
cache=31
checkpoints=0
checksum=uncompressed
chunk_size=7
compaction=0
compression=lz4
data_extend=0
data_source=table
delete_pct=36
dictionary=0
evict_max=2
file_type=row-store
backups=0
huffman_key=0
huffman_value=0
insert_pct=28
internal_key_truncation=1
internal_page_max=12
isolation=random
key_gap=3
key_max=93
key_min=19
leak_memory=0
leaf_page_max=17
logging=0
logging_archive=1
logging_prealloc=1
lsm_worker_threads=4
merge_max=15
mmap=1
ops=100000
prefix_compression=1
prefix_compression_min=7
repeat_data_pct=60
reverse=0
rows=100000
runs=1
split_pct=77
statistics=1
statistics_server=0
threads=15
timer=20
value_max=1471
value_min=8
wiredtiger_config=
write_pct=53
############################################
```

Maybe we need to use the decompress_safe API rather than the decompress_fast API (https://github.com/Cyan4973/lz4/blob/master/lib/lz4.h)? in LZ4 to avoid this?

Failure:

http://build.wiredtiger.com:8080/job/wiredtiger-test-format-stress-sanitizer/2995/console

**Comment by agorrod** [comment]
While attempting to reproduce this failure I saw something different:

```
  29: table, row-store (6 seconds)                                            
[1427074119:447173][92806:0x7fac0d34a740], t, file:wt.wt, session.verify: lz4 error: LZ4 decompress error: -17437
[1427074119:447204][92806:0x7fac0d34a740], t, file:wt.wt, session.verify: cell 142 on page at [3223552-3254272, 30720, 2975935079] references an overflow item at [2262528-2263040, 512, 1810685079] that failed verification: WT_ERROR: non-specific WiredTiger error
[1427074119:449827][92806:0x7fac0d34a740], t, file:wt.wt, session.verify: checkpoint ranges never verified: 12
[1427074119:449870][92806:0x7fac0d34a740], t, file:wt.wt, session.verify: file ranges never verified: 12
t: session.verify: table:wt: post-corrupt-salvage verify: WT_ERROR: non-specific WiredTiger error
```

Appears as though salvage thinks it worked, but didn't create a valid database.

**Comment by agorrod** [comment]
This wasn't a once off:
http://build.wiredtiger.com:8080/job/wiredtiger-test-format-stress/6450/changes

**Comment by keithbostic** [comment]
Closed with the merge of bea03a7.
